gamemodes/terrortown/gamemode/shared/sh_sprint.lua

```diff
@@ -87,11 +87,16 @@ hook.Add("Think", "TTT2PlayerSprinting", function()
 		if not ply.sprintTS then continue end
 
 		local timeElapsed = CurTime() - ply.sprintTS
-
-		if not ply.sprintMultiplier then
-			ply.sprintProgress = math.min((ply.oldSprintProgress or 0) + timeElapsed * GetGlobalFloat("ttt2_sprint_stamina_regeneration"), 1)
+		local isSprinting = (ply.sprintMultiplier ~= nil)
+
+		local drainMult, regenMult = hook.Run("TTT2StaminaMult", ply, isSprinting, ply.sprintProgress)
```

Comment:
it is right to send data with this hook, but should think about consistency. isSprinting and ply.sprintProgess is accessible on the submitted Player "object" "ply" itself.

Reply:
isSprinting should probably be kept - I think it's not intuitive to potential addon developers, and providing the information makes sense.

Reply:
Hmm... It's okay if we handle it this way...